### DISCUSSION TITLE
Teensy 4.1 Multicast

### DISCUSSION THREAD

**Discussion by g0orx** [ssilverman/QNEthernet#76] [Q&A] [answered]
I have been trying to get Multicast working on a Teensy 4.1. I get a failure when I try to join a group. I changed the code to set errno to the error returned from igmp_joingroup_netif so that I could print out the error which is -6 (ENXIO - No such device or address).  I am using DHCP to get the address of the Teensy and then use Ethernet.broadcastIP() to get the IP address to pass as the group IP address.

Udp.beginMulticastWithReuse(Ethernet.broadcastIP(), 1024)

Some debug output showing DHCP worked OK:

> Initialize Ethernet
> Ethernet begin
> Get DHCP address
> linkstate=true
> IP Address: 192.168.1.124
> Broadcast Address: 192.168.1.255
> Udp.beginMulticast failed: errno=-6

Any ideas why this is not working? Do I need to change some configuration in QNEthernet.

**Comment by ssilverman** [comment, marked as answer]
The broadcast address isn’t a multicast address.

**Reply by g0orx** [reply]
Thanks. I was being stupid. Now works OK with a valid Multicast address.

**Reply by ssilverman** [reply]
Great! I’m glad it’s working.
